src/notebook/plugin.ts

```diff
@@ -40,20 +51,66 @@ export
 function resolve(container: Container): void {
   container.resolve({
     requires: [IServicesProvider, IFileOpener],
-    create: (services, opener) => {
-      let handler = new NotebookFileHandler(services.contentsManager);
+    create: (services: IServicesProvider, opener: IFileOpener) => {
+      let handler = new NotebookFileHandler(services.contentsManager, services.notebookSessionManager);
       opener.register(handler);
     }
   });
 }
 
+export
+class SessionStoreMapping {
+  constructor(services: IServicesProvider) {
+    this.services = services;
+  }
+  public services: IServicesProvider;
+}
+
+function messageToModel(msg: IKernelMessage) {  
+  let m: Output = msg.content;
+  let type = msg.header.msg_type;
+  if (type === 'execute_result') {
+    m.output_type = 'display_data';
+  } else {
+    m.output_type = type;
+  }
+  return buildOutputModel(m);
+}
+
+function executeSelectedCell(model: NotebookModel, session: INotebookSession)  {
+  let cell = model.cells.get(model.selectedCellIndex);
+  if (isCodeCell(cell)) {
+    let exRequest = {
+      code: cell.input.textEditor.text
+    };
+    let output = cell.output;
+    console.log(`executing`, exRequest)
+    let ex = session.kernel.execute(exRequest);
+    output.clear(false);
+    ex.onIOPub = (msg => {
+      // TODO: not getting an execute_result message
```

Comment:
Fixed it, the following options must be passed to execute to mimic what is done in the current notebook:

```
{
            silent: false,
            store_history: true,
            stop_on_error: true,
            allow_stdin: true,
        };
```